src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -505,15 +505,19 @@ def add(self,
 
     def remove(self,
                level_spec: LevelSpec,
-               interval: Optional[str],
-               start_time: Optional[str]) -> None:
+               interval_str: Optional[str],
+               start_time_str: Optional[str]) -> None:
+        if interval_str:
+            interval = Interval.from_string(interval_str)
+        else:
+            interval = None
```

Comment:
Do we need this if/else clause? According to the function signature `interval_str` may not be `None`, and for an empty string I suppose `Interval.from_string` will rise "Invalid interval" error as expected?

Reply:
And this `interval = None` makes jenkins fail:
```
rbd_support/schedule.py: note: In member "remove" of class "Schedules":
rbd_support/schedule.py:513: error: Incompatible types in assignment (expression has type "None", variable has type "Interval")
```

Reply:
Yes, without it results in Traceback in Line 236 in `Interval.from_string `. (TypeError: expected string or bytes-like object).

Reply:
So `interval_str` is not actually `str` in that case but `None`? Strange. Do you have a reproducer to try?

Reply:
True, Line 106 `module.py` in `mirror_snapshot_schedule_remove`  if we omit `interval` to remove  command with pool name, it assigns `None`.

Reply:
Ah. I see now. Indeed. Anyway,  `interval = None` causes mypy error I posted above, and it should be improved somehow. Interesting, if you change your code to `interval = Interval.from_string(interval_str) if interval_str else None` will it still fail?

Reply:
then, we will have `referenced before assignment` at line 517.

Reply:
I think we can add  `None` handling to `Interval.from_string` similarly to what we have in `StartTime.from_string` and you will not need any condition on this level. I was reluctant to suggest this from the start, because it does not look entirely correct to me and might hide an issue that will be difficult to catch at runtime. But taking we already have similar for `StartTime` I will be ok with this.

Unless someone more experienced with python comes with a better suggestion.